src/forgd_core/curves/single/exponential.py

```
@@ -197,7 +215,11 @@ def buy(self, request: 'TransactionRequest') -> 'TransactionResult':
         if not self.options["allow_buy"]:
             raise ValueError("Buys are disabled on this exponential bonding curve.")
 
-        original_amount = request.amount
+        original_amount = round_to_tick(
```

After thinking about it, i think we may want to throw an exception if the caller does not specify a correct decimal places order request?

If they get less than they request it may create subtle issues that the caller/UI wont notice initially.

If we force that the qty must be correct (throw exception otherwise) we are forcing the caller/UI to deal with the issue up front.

Let me know what you think.

---

I agree we should have some catches for request. 

Will add that to the project. Let me know if you can complete.